### PR TITLE
Updates to address items listed in issue 63

### PR DESCRIPTION
**Pull request by mkavulich**
- Clean up Glossary and linked terms
 - Replace mentions of forum with GitHub discussions
 - Updates to host side coding based on changes to CCPP Physics
 - Update Code Management for new Application Fork workflow
 - Some miscellaneous fixes

You can view the latest version of this branch's proposed changes in ReadTheDocs here: https://mike-k-tech-doc.readthedocs.io/en/latest/index.html

### CODE REVIEW

File: CCPPtechnical/source/Overview.rst [hunk 1]
```diff
@@ -101,7 +101,7 @@ This documentation is housed in repository https://github.com/NCAR/ccpp-doc.
 
 The CCPP is governed by the groups that contribute to its development. The governance
```
Comment (ligiabernardet) [inline CCPPtechnical/source/Overview.rst:102]: I suggest a change for the CCPP Physics governance and code mgmt: 
The CCPP Physics code management is collaboratively determined by NOAA, NCAR, and the Navy Research Laboratory (NRL), and the DTC works with EMC and its sponsors to determine :term:`schemes <scheme>` and suites to be included and supported.

File: CCPPtechnical/source/Overview.rst [hunk 2]
```diff
@@ -208,8 +208,10 @@ Additional Resources
 For the latest version of the released code and additional documentation,
 please visit the `DTC Website <https://dtcenter.org/community-code/common-community-physics-package-ccpp>`_.
 
-Please send questions and comments to the CCPP Forum at https://dtcenter.org/forum/ccpp-user-support.
-When using the CCPP with the UFS, questions can also be posted
-in the UFS Forum at https://forums.ufscommunity.org/.
+Please post questions and comments to the GitHub discussions board for the relevant code repository:
+ - *CCPP Physics* https://github.com/NCAR/ccpp-physics/discussions
```
Comment (ligiabernardet) [inline CCPPtechnical/source/Overview.rst:212]: Is there a reason why the first two bullets are in italics and the remaining ones are not?

File: CCPPtechnical/source/ScientificDocRules.inc [hunk 3]
```diff
@@ -131,7 +131,7 @@ or ``@ref``. Example from ``suite_FV3_GFS_v16.txt``:
 
    Version 16 of the Global Forecast System (GFS) was implemented operationally by the NOAA
    National Centers for Environmental Prediction (NCEP) in 2021. This suite is available for
-   use with the UFS SRW App and with the CCPP SCM.
+   use with the :term:`UFS` SRW App and with the CCPP :term:`SCM`.
```
Comment (ligiabernardet) [inline CCPPtechnical/source/ScientificDocRules.inc:134]: Not rendering correctly

File: CCPPtechnical/source/CompliantPhysicsParams.rst [hunk 4]
```diff
@@ -84,12 +84,12 @@ General Rules
 =============
 A CCPP-compliant scheme is written in the form of Fortran modules. Each scheme must be in its own module, and must include at least one of the
 following subroutines (*entry points*): *_init*, *_timestep_init*, *_run*, *_timestep_finalize*,
-and *_finalize*. Each subroutine corresponds to one of the five *phases* of the CCPP framework as described above.
+and *_finalize*. Each subroutine corresponds to one of the five *phases* of the :term:`CCPP framework` as described above.
```
Comment (ligiabernardet) [inline CCPPtechnical/source/CompliantPhysicsParams.rst:87]:
```suggestion
and *_finalize*. Each subroutine corresponds to one of the five *phases* of the :term:`CCPP Framework` as described above.
```

**Review by ligiabernardet — CHANGES_REQUESTED** [review]
Thank you for the fixes and upgrades, as well as for adding many more links to the definitions in the glossary. Looks great.
The acronym NEMSfv3gfs is no longer used and should be removed from the list of acronyms.

File: CCPPtechnical/source/CompliantPhysicsParams.rst [hunk 5]
```diff
@@ -37,7 +37,7 @@ The implementation of a driver is reasonable under the following circumstances:
   and rearranging the index order, for example, ``cu_gf_driver.F90`` or ``gfdl_cloud_microphys.F90``
   in the ``ccpp-physics/physics`` directory.
 
-Schemes in the CCPP are classified into two categories: *primary* schemes and *interstitial* schemes.
+Schemes in the CCPP are classified into two categories: :term:`*primary* schemes <primary scheme>` and :term:`*interstitial* schemes <interstitial scheme>`.
```
[the remark this answers is not in the document]
Comment (mkavulich) [reply CCPPtechnical/source/CompliantPhysicsParams.rst:40]: Unfortunately there's a limitation that the rst specification does not allow for nested formatting. I wanted some way to emphasize the difference between these two scheme types, but perhaps it is just better to let the links speak for themselves.

**Comment by mkavulich** [comment]
@ligiabernardet Thank you for the thorough reivew! I believe I have addressed all the changes you requested, let me know if you have more comments.

**Review by ligiabernardet — APPROVED** [review]
Thanks for the updates.
The only think I did not see addressed was the removal of NEMSfv3gfs from the list of acronyms.

**Review by grantfirl — APPROVED** [review]
The changes all look OK to me.

**Pull request merged**